lib/sparkql/parser.y

```diff
@@ -52,77 +52,99 @@ rule
     ;
 
 ##### Expression
-# 
-# The core of the filtering system, the expression requires a field, a condition 
-# and criteria for comparing the value of the field to the value(s) of the 
-# condition. The result of evaluating the expression on a resource is a true of 
+#
+# The core of the filtering system, the expression requires a field, a condition
+# and criteria for comparing the value of the field to the value(s) of the
+# condition. The result of evaluating the expression on a resource is a true of
 # false for matching the criteria.
   expression
-    : field OPERATOR condition { result = tokenize_expression(val[0], val[1],val[2]) }
-    | field RANGE_OPERATOR range { result = tokenize_expression(val[0], val[1], val[2]) }
+    : field_expression OPERATOR condition { result = tokenize_expression(val[0], val[1],val[2]) }
+    | field_expression RANGE_OPERATOR range { result = tokenize_expression(val[0], val[1], val[2]) }
     | group
     ;
-  
+
 ##### Unary Conjunction
-# 
-# Some conjunctions don't need to expression at all times (e.g. 'NOT'). 
+#
+# Some conjunctions don't need to expression at all times (e.g. 'NOT').
   unary_conjunction
     : UNARY_CONJUNCTION expression { result = tokenize_unary_conjunction(val[0], val[1]) }
-    ;  
-  
+    ;
+
 ##### Conjunction
-# 
+#
 # Two expressions joined together using a supported conjunction
   conjunction
     : expressions CONJUNCTION expression { result = tokenize_conjunction(val[0], val[1],val[2]) }
     | expressions UNARY_CONJUNCTION expression { result = tokenize_conjunction(val[0], val[1],val[2]) }
     ;
-  
+
 ##### Group
-# 
+#
 # One or more expressions encased in parenthesis. There are limitations on nesting depth at the time of this writing.
+
   group
-  	: LPAREN expressions RPAREN { result = tokenize_group(val[1]) }
-  	;
+    : LPAREN expressions RPAREN { result = tokenize_group(val[1]) }
+    ;
 
 ##### Field
-# 
-# Keyword for searching on, these fields should be discovered using the metadata 
-# rules. In general, Keywords that cannot be found will be dropped from the 
+#
+# Keyword for searching on, these fields should be discovered using the metadata
+# rules. In general, Keywords that cannot be found will be dropped from the
 # filter.
 
+  field_expression
+    : field_function_expression
+    | arithmetic_expression
+    ;
+
   field
-    : field ADD field { result = tokenize_arithmetic(val[0], val[1], val[2]) }
-    | field SUB field { result = tokenize_arithmetic(val[0], val[1], val[2]) }
-    | field MUL field { result = tokenize_arithmetic(val[0], val[1], val[2]) }
-    | field DIV field { result = tokenize_arithmetic(val[0], val[1], val[2]) }
-    | field MOD field { result = tokenize_arithmetic(val[0], val[1], val[2]) }
-    | STANDARD_FIELD
+    : STANDARD_FIELD
     | CUSTOM_FIELD
+    ;
+
+  arithmetic_expression
```

Comment:
I'd rename this to have the word `field` in it. The point being is we can have arithmetic with the field or arithmetic with the condition, and never the twain shall meet.

Reply:
Good call 👍